circuit_tutorials/plonky2/merkle_tree/src/main.rs

```diff
@@ -0,0 +1,238 @@
+        .multipart(upload)
+        .send()
+        .await
+        .unwrap();
+    assert_eq!(
+        &response.status().as_u16(),
+        &201u16,
+        "Expected status code 201"
+    );
+    let response_body = response.json::<Value>().await.unwrap();
+    let circuit_id = response_body["circuit_id"].as_str().unwrap();
+    println!("Circuit ID: {:?}", &circuit_id);
+
+    // Poll circuit detail until it has a status of Ready or Failed
+    let circuit_data = poll_status(
+        format!("circuit/{circuit_id}/detail"),
+        &api_url,
+        &api_key,
+        600,
+    )
+    .await;
+    if circuit_data["status"].as_str().unwrap().contains("Failed") {
+        println!("Circuit compilation failed.");
+        std::process::exit(1);
+    }
+
+    println!("Saving circuit details locally");
+    std::fs::create_dir_all("./data").unwrap();
+    let file = File::create("./data/compile_out.json").unwrap();
+    let mut writer = BufWriter::new(file);
+    serde_json::to_writer_pretty(&mut writer, &circuit_data).unwrap();
+    writer.flush().unwrap();
+}
+
+async fn prove_circuit(json_input_path: &str) {
+    {
+        dotenv().expect("Failed to read .env file");
+        let api_key: String = std::env::var("SINDRI_API_KEY").unwrap();
+        let api_url_prefix: &str = "https://sindri.app/api/";
+
+        let api_version: &str = "v1/";
+        let api_url: String = api_url_prefix.to_owned() + api_version;
+
+        println!("Reading circuit details locally");
+        let mut file = File::open("./data/compile_out.json").unwrap();
+        let mut data = String::new();
+        file.read_to_string(&mut data).unwrap();
+        let circuit_data: Value = serde_json::from_str(&data).unwrap();
+        let circuit_id = &circuit_data["circuit_id"].as_str().unwrap();
+        let circuit_id = circuit_id;
+
+        // Initiate proof generation.
+        println!("Reading proof input from input.json file");
+        let mut proof_input_file = File::open(json_input_path).unwrap();
+        let mut proof_input = String::new();
+        proof_input_file.read_to_string(&mut proof_input).unwrap();
+        let map = serde_json::json!({"proof_input": proof_input});
+
+        println!("Requesting a proof");
+        let client = Client::new();
+        let response = client
+            .post(format!("{api_url}circuit/{circuit_id}/prove"))
+            .headers(headers_json(&api_key))
+            .json(&map)
+            .send()
+            .await
+            .unwrap();
+        assert_eq!(
+            &response.status().as_u16(),
+            &201u16,
+            "Expected status code 201"
+        );
+        let response_body = response.json::<Value>().await.unwrap();
+        let proof_id = response_body["proof_id"].as_str().unwrap();
+
+        // Poll proof detail until it has a status of Ready or Failed
+        let proof_data =
+            poll_status(format!("proof/{proof_id}/detail"), &api_url, &api_key, 600).await;
```

Comment:
I'd prefer that `prove_circuit` function doesn't define the path to poll for status. I suggest adding `poll_proof_status` and `poll_circuit_status` which themselves define some of the defaults and the path, and in turn call `poll_status` with those arguments.

Reply:
done [b61e566](https://github.com/Sindri-Labs/sindri-resources/pull/98/commits/b61e566afbdfe36cadaab0ef40c062bfceaa1dcc)